src/diagnostics/logPointsClient.ts

```diff
@@ -76,9 +77,15 @@ export class KuduLogPointsDebuggerClient extends LogPointsDebuggerClientBase imp
         return buf.toString('base64');
     }
 
-    public async startSession(siteName: string, affinityValue: string, publishCredential: WebSiteModels.User): Promise<CommandRunResult<IStartSessionResponse>> {
+    public async startSession(siteName: string, affinityValue: string, publishCredential: WebSiteModels.User, data: IStartSessionRequest): Promise<CommandRunResult<IStartSessionResponse>> {
         await this.uploadSshClient(siteName, affinityValue, publishCredential);
-        return this.makeCallAndLogException<IStartSessionResponse>(siteName, affinityValue, publishCredential, "curl -s -S -X POST http://localhost:32923/debugger/session");
+        const jsonPayloadBase64 = KuduLogPointsDebuggerClient.base64Encode({
+            username: data.username
```

Comment:
what is `data.username`?  Is this the user that you're auth'd against azure?

Reply:
Are you  sure it's not supposed to be publishCredential.userName?

Reply:
`data.username` is the email address of the active Azure tenant identity.

I don't remember what publishCredential.userName is like. It will at most be the same as userId, so userId is always better to use for our purpose. We want to have a unique identifier for the current logged in user to differentiate "who" initiated the debug session.